runner/operator/access/v1_0_0/msi/__init__.py

```diff
@@ -0,0 +1,145 @@
+"""""""""""""""""""""""""""""
+" ACCESS-Pipeline MSI workflow operator
+" http://www.github.com/mskcc/access-pipeline/workflows/msi.cwl
+"""""""""""""""""""""""""""""
+
+import os
+import json
+from jinja2 import Template
+
+from runner.models import Port, Run, RunStatus
+from file_system.models import FileMetadata
+from runner.operator.operator import Operator
+from runner.serializers import APIRunCreateSerializer
+from file_system.repository.file_repository import File, FileRepository
+
+
+WORKDIR = os.path.dirname(os.path.abspath(__file__))
+
+class AccessLegacyMSIOperator(Operator):
+
+    # Will be populated after get_sample_inputs()
+    sample_inputs = None
+    number_of_inputs = None
+
+    def get_sample_inputs(self):
+        """
+        Create all sample inputs for all runs triggered in this instance of the operator
+
+        :return: list of json_objects
+        """
+        access_duplex_output_ports = Port.objects.filter(
+            name='duplex_bams',
+            run__app__name='access legacy',
+            run__status=RunStatus.COMPLETED
+        )
+        # Each port is a list, so need a double list comprehension here
+        all_access_output_records = [f for p in access_duplex_output_ports for f in p.value]
+        # these are port objects, they dont have metadata field
+        all_access_completed_samples = [r['sampleId'] for r in all_access_output_records]
```

Comment:
I'll try to find an example of the collapsing pipeline outputs and how the ports for that look after a completed run, so you can understand how these "record" objects work

Reply:
@aef- So, instead of this^, you can refer to how I'm doing it here:

https://github.com/mskcc/beagle/pull/599/files#diff-7f514a1f5bc70172b2c82eb7b6da8b76dc955eb056b917269abd4164bc25bb4eR28

Reply:
I'm giving up on the record outputs because I realized they are showing up as "null", so it probably makes sense to do it this way, and just use "-L0" and "_cl_aln" from the file names to get the T/N and sample ID information